File: src/codes/hardcoded-variability/php/uncode.php
<?php

class NotificationService {
  public function sendEmail(string $user, string $message): void {
    echo "件名: お知らせ\n";
    echo "本文: {$message}\n";
    echo "宛先: {$user}@example.com\n";
  }

  public function sendSlack(string $user, string $message): void {
    echo "Slack宛: {$user}\n";
    echo "内容: {$message}\n";
  }
}

// 利用例
$notifier = new NotificationService();
$notifier->sendEmail("hiroshi", "メッセージ1");
$notifier->sendSlack("hiroshi", "メッセージ2");

File: src/codes/hardcoded-variability/py/uncode.py
class NotificationService:
    def send_email(self, user: str, message: str):
        print("件名: お知らせ")
        print(f"本文: {message}")
        print(f"宛先: {user}@example.com")

    def send_slack(self, user: str, message: str):
        print(f"Slack宛: {user}")
        print(f"内容: {message}")

# 利用例
notifier = NotificationService()
notifier.send_email("hiroshi", "メッセージ1")
notifier.send_slack("hiroshi", "メッセージ2")
